### PR TITLE
Upgrade to C65

### PR DESCRIPTION
**Pull request by hferreiro**

### CODE REVIEW

File: atom/browser/api/atom_api_web_contents.cc [hunk 1]
```diff
@@ -1366,20 +1365,6 @@ void WebContents::DidFailLoad(content::RenderFrameHost* render_frame_host,
       is_main_frame);
 }
 
-void WebContents::DidGetResourceResponseStart(
```
Comment (hferreiro) [inline atom/browser/api/atom_api_web_contents.cc]: @bridiver this change requires and alternative for browser-laptop. From https://bugs.chromium.org/p/chromium/issues/detail?id=783981, `WebContentsObserver::DidFinishNavigation` is suggested as an alternative.

File: atom/browser/api/atom_api_web_contents.cc [hunk 2]
```diff
@@ -1760,9 +1760,7 @@ bool WebContents::IsLoading() const {
 bool WebContents::IsLoadingMainFrame() const {
   // Comparing site instances works because Electron always creates a new site
   // instance when navigating, regardless of origin. See AtomBrowserClient.
-  return (web_contents()->GetLastCommittedURL().is_empty() ||
-          web_contents()->GetSiteInstance() !=
-          web_contents()->GetPendingSiteInstance()) && IsLoading();
+  return web_contents()->IsLoadingToDifferentDocument();
```
Comment (hferreiro) [inline atom/browser/api/atom_api_web_contents.cc]: @bridiver I was about to ask what tests to perform to check if the new code is equivalent, but I see it isn't used at all in `browser-laptop`, can we remove it?

File: atom/common/native_mate_converters/v8_value_converter.cc [hunk 3]
```diff
@@ -263,23 +263,36 @@ base::Value* V8ValueConverter::FromV8ValueImpl(
...
+  // // If val was created in a different context than our current one,
+  // // change to
+  // // that context, but change back after val is converted.
+  // if (!val->CreationContext().IsEmpty() &&
+  //     val->CreationContext() != isolate->GetCurrentContext()) {
+  //   context = val->CreationContext();
+  //   scope.reset(new v8::Context::Scope(context));
+  // } else {
+    context = isolate->GetCurrentContext();
+  // }
```
Comment (hferreiro) [inline atom/common/native_mate_converters/v8_value_converter.cc]: @bridiver was this check omitted on purpose?

File: atom/common/native_mate_converters/v8_value_converter.cc [hunk 4]
```diff
@@ -340,17 +357,22 @@ base::Value* V8ValueConverter::FromV8Array(
     return new base::Value();
 
   std::unique_ptr<v8::Context::Scope> scope;
+  v8::Local<v8::Context> context;
   // If val was created in a different context than our current one, change to
   // that context, but change back after val is converted.
```
Comment (hferreiro) [inline atom/common/native_mate_converters/v8_value_converter.cc]: @bridiver the context isn't restored at all, is it a pending task?

File: atom/browser/api/atom_api_window.cc [hunk 5]
```diff
@@ -784,9 +784,15 @@ bool Window::IsModal() const {
 }
 
 v8::Local<v8::Value> Window::GetNativeWindowHandle() {
```
Comment (hferreiro) [inline atom/browser/api/atom_api_window.cc]: This function is exposed in BrowserWindow but it isn't use at all by `browser-laptop`, can we remove it?

File: atom/app/atom_content_client.cc [hunk 6]
```diff
@@ -116,9 +116,9 @@ AtomContentClient::AtomContentClient() {
 AtomContentClient::~AtomContentClient() {
 }
 
-void AtomContentClient::SetActiveURL(const GURL& url) {
-  base::debug::SetCrashKeyValue(crash_keys::kActiveURL,
-                                url.possibly_invalid_spec());
```
[the remark this answers is not in the document]
Comment (hferreiro) [reply atom/app/atom_content_client.cc]: This method isn't called anywhere, do you mean we need to do that?

[the remark this answers is not in the document]
Comment (darkdh) [reply atom/app/atom_content_client.cc]: I mean we have to do this
```
base::debug::SetCrashKeyValue(crash_keys::kActiveURL,
                              url.possibly_invalid_spec());
static crash_reporter::CrashKeyString<64> top_origin_key("top-origin");
top_origin_key.Set(top_origin);
```
you can't just discard the line setting crash key for url no matter we are calling it or not

File: brave/browser/component_updater/widevine_cdm_component_installer.cc [hunk 7]
```diff
@@ -212,6 +219,32 @@ std::string GetCodecs(const base::DictionaryValue& manifest) {
...
+                             base::SPLIT_WANT_NONEMPTY);
+
+  for (const auto& codec : supported_codecs) {
+    if (codec == kCdmSupportedCodecVp8)
+      supported_video_codecs.push_back(media::VideoCodec::kCodecVP8);
+    else if (codec == kCdmSupportedCodecVp9)
+      supported_video_codecs.push_back(media::VideoCodec::kCodecVP9);
+#if BUILDFLAG(USE_PROPRIETARY_CODECS)
+    else if (codec == kCdmSupportedCodecAvc1)
+      supported_video_codecs.push_back(media::VideoCodec::kCodecH264);
```
Comment (darkdh) [inline brave/browser/component_updater/widevine_cdm_component_installer.cc]: we should also make change of `IsWidevineAvailable` in `atom/app/atom_content_client.cc` which still uses `kCdmSupportedCodec*`

File: brave/browser/component_updater/widevine_cdm_component_installer.cc [hunk 8]
```diff
@@ -405,10 +457,11 @@ void WidevineCdmComponentInstallerPolicy::RegisterWidevineCdmWithBrave(
     std::unique_ptr<base::DictionaryValue> manifest) {
   DCHECK_CURRENTLY_ON(BrowserThread::UI);
   const std::string codecs = GetCodecs(*manifest);
+  bool supports_persistent_license = GetPersistentLicenseSupport(*manifest);
 
   content::PepperPluginInfo plugin_info;
   if (!MakeWidevineCdmPluginInfo(cdm_version, cdm_install_dir, codecs,
-                                 &plugin_info)) {
+                                 supports_persistent_license, &plugin_info)) {
```
Comment (darkdh) [inline brave/browser/component_updater/widevine_cdm_component_installer.cc]: This change (`supports_persistent_license`) should also apply to `AtomContentClient::AddContentDecryptionModules`

File: atom/app/atom_content_client.cc (hunk 6 above)
[the remark this answers is not in the document]
Comment (hferreiro) [reply atom/app/atom_content_client.cc]: `base::debug::SetCrashKeyValue()` is the deprecated API that this commit is updating. The crash key is set with the new code.

[the remark this answers is not in the document]
Comment (darkdh) [reply atom/app/atom_content_client.cc]: right, I mean you have to do something for url arg like
```
static crash_reporter::CrashKeyString<1024> active_url("url-chunk");
active_url.Set(url.possibly_invalid_spec());
```
instead of completely delete the lines and not using it

File: brave/browser/brave_content_browser_client.cc [hunk 9]
```diff
@@ -359,22 +359,6 @@ void BraveContentBrowserClient::BindInterfaceRequestFromFrame(
...
 }
 
-bool BraveContentBrowserClient::BindAssociatedInterfaceRequestFromFrame(
-    content::RenderFrameHost* render_frame_host,
-    const std::string& interface_name,
-    mojo::ScopedInterfaceEndpointHandle* handle) {
-  // TODO(https://crbug.com/736357): Factor AssociatedInterfaceRegistryImpl out
-  // into content/public/ so it can be used here instead of this abomination.
-  if (interface_name == password_manager::mojom::CredentialManager::Name_) {
-    BravePasswordManagerClient::BindCredentialManager(
```
[the remark this answers is not in the document]
Comment (hferreiro) [reply brave/browser/brave_content_browser_client.cc]: Pushed a fixup.

File: brave/browser/component_updater/widevine_cdm_component_installer.cc (hunk 7 above)
Comment (hferreiro) [reply brave/browser/component_updater/widevine_cdm_component_installer.cc]: Pushed a fixup.

File: brave/browser/component_updater/widevine_cdm_component_installer.cc (hunk 8 above)
Comment (hferreiro) [reply brave/browser/component_updater/widevine_cdm_component_installer.cc]: Pushed a fixup.

File: atom/browser/api/atom_api_web_contents.cc (hunk 1 above)
Comment (darkdh) [reply atom/browser/api/atom_api_web_contents.cc]: can you use the alternative to emit the same event `did-get-response-details` to browser-laptop?

Comment (hferreiro) [reply atom/browser/api/atom_api_web_contents.cc]: I have pushed a commit here: https://github.com/brave/muon/tree/did-get-response-details
It breaks on startup because `GetResponseHeaders()` returns `null` sometimes, any idea about what to do in those cases?

Comment (darkdh) [reply atom/browser/api/atom_api_web_contents.cc]: https://github.com/brave/muon/pull/537/commits/4b1636c209b6bb27e7fb8b5d96bdd4fe2d98010a
and you also missed emitting headers

File: patches/master_patch.patch [hunk 10]
```diff
@@ -1951,21 +1916,6 @@ index 7505524b693b24d9028447929d881232a0483561..03eea7522154c40691405ec43e07303b
...
                report.uuid, Metrics::CrashSkippedReason::kUploadThrottled);
-diff --git a/third_party/widevine/cdm/BUILD.gn b/third_party/widevine/cdm/BUILD.gn
-index f35d8187209663167a19a06a745b95a89d7b9063..36f7959a0508676832b8ecd8d1912fce7456f6e2 100644
---- a/third_party/widevine/cdm/BUILD.gn
-+++ b/third_party/widevine/cdm/BUILD.gn
-@@ -110,6 +110,10 @@ if (widevine_cdm_binary_files != []) {
-       "//build/config:exe_and_shlib_deps",
-     ]
- 
-+    if (is_mac) {
```
Comment (bridiver) [inline patches/master_patch.patch]: why was this removed?

Comment (hferreiro) [reply patches/master_patch.patch]: it's already upstream https://crrev.com/232a383ba687

Comment (bridiver) [reply patches/master_patch.patch]: great, they finally fixed their broken build :)
